Sources/MapboxDirections/RouteStep.swift

```diff
@@ -558,21 +626,53 @@ open class RouteStep: Codable {
     
     
     /// Used to Decode `Intersection.admin_index`
-    private struct AdministrativeAreaIndex: Codable {
+    private struct AdministrativeAreaIndex: Codable, ForeignMemberContainer {
+        var foreignMembers: JSONObject = [:]
```

Comment:
As you pointed out in chat, this struct also contributes to the performance hit. I believe this is because it’s an artificial construct that redecodes Intersection, which is already a foreign member container. The `AdministrativeAreaIndex.foreignMembers` property would end up containing a duplicate copy of the whole intersection object.